cinema/views.py

```diff
@@ -1,11 +1,14 @@
 from datetime import datetime
 
 from django.db.models import F, Count
-from rest_framework import viewsets
+from rest_framework import viewsets, mixins
 from rest_framework.pagination import PageNumberPagination
+from rest_framework.permissions import IsAuthenticated
```

Comment:
Ensure that the custom permissions used in your project, such as `IsAdminOrIfAuthenticatedReadOnly`, are correctly imported and defined. Currently, only `IsAuthenticated` is imported.